File: insertion_Sort.py
# select one element at a time from the left of the collection.
# insert the element at the proper position
# after insertion of the element to the left is sorted.


def insertionSort(A):
    l=len(A)
    for i in range(1,l):
        cvalue=A[i]
        pos=i
        while pos>0 and A[pos-1]>cvalue:
            A[pos]=A[pos-1]
            pos=pos-1
        A[pos]=cvalue

A=[3,5,7,2,1,8]
print("original:",A)

insertionSort(A)
print("sorted:",A)

File: insertion_sort.go
package main

import "fmt"

func insertionSort(A []int) []int {
	l := len(A)
	for i := 1; i < l; i++ {
		cvalue := A[i]
		pos := i
		// fmt.Println(cvalue, pos, A[pos])
		for pos > 0 && A[pos-1] > cvalue {
			A[pos] = A[pos-1]
			// fmt.Println("vinod")
			pos = pos - 1
		}
		A[pos] = cvalue
	}

	return A
}

func main() {
	A := []int{3, 5, 7, 2, 1, 8}
	fmt.Println("original:", A)
	insertionSort(A)
	fmt.Println(A)
}
